Fixing help popup covering company popup

spirani opened the pull request:
Hi, I was having the same issue as #18.

In this pull request, the ```company-quickhelp--show``` function was changed to prevent the help popup from covering up the company completions list.

Before:
![before](https://cloud.githubusercontent.com/assets/4155393/8693496/8ef43126-2aa5-11e5-8e50-46ff31b692af.png)

After:
![after](https://cloud.githubusercontent.com/assets/4155393/8693497/92d0a054-2aa5-11e5-9886-d74904eaeee7.png)

expez commented:
Thanks!  

A while back, this is what `company-quickhelp--show` looked like:

```el
(defun company-quickhelp--dx-offset ()
  "Offset to ensure the quickhelp popup doesn't appear above
completion candidates."
  (let* ((width (overlay-get ovl 'company-width))
         (col (overlay-get ovl 'company-column))
         (extra (- (+ width col) (company--window-width))))
    (* (frame-char-width)
       (- width (length company-prefix)
          (if (< 0 extra) extra 1)))))

(defun company-quickhelp--show ()
  (company-quickhelp--cancel-timer)
  (let* ((selected (nth company-selection company-candidates))
         (doc (company-quickhelp--doc selected))
         (ovl company-pseudo-tooltip-overlay)
         (x-gtk-use-system-tooltips nil))
    (when (and ovl doc)
      (with-no-warnings
        (pos-tip-show doc
                      nil
                      nil
                      nil
                      300
                      80
                      nil
                      (company-quickhelp--dx-offset))))))
```

In #2 someone pointed out that in certain situations--I could never figure out why--this offset caused the popup to be placed way to the right of the completion candidates.  I'm afraid we might get into the same situation here, so I'd like to test this for a while before merging it in.

expez commented:
I could only reproduce the problem in #2 by using `elpy` to edit some python code and when I try this new code using `elpy` the popup is misaligned, but it's an improvement on the old behavior which is also misaligned heh

spirani commented:
Yes, more testing definitely a good idea.

When using elpy, the popup is misaligned for me too. Does it look similar for you as well?

![align1](https://cloud.githubusercontent.com/assets/4155393/8708241/df7dd53a-2b08-11e5-94c1-8561494b60a8.png)

For me, this only happens when editing a line at the end of the file.
However, if I insert a few newlines at the end and go back to where I was before, it looks fine:

![align2](https://cloud.githubusercontent.com/assets/4155393/8708276/05a584f6-2b09-11e5-819a-ce96c94f5ff6.png)
This vertical misalignment seems to happen both with and without this code.

expez commented:
> Does it look similar for you as well?

Yes, this is what I saw yesterday.  It was either blocking the candidates or misaligned vertically.  Of the two this latter behavior is certainly the lesser of two evils.

> However, if I insert a few newlines at the end and go back to where I was before, it looks fine

Yeah, I wish I knew what caused this difference, but this also means it's going to work OK in most cases.

I think it's time to merge this into master, if you'll humor me with a few changes:

* Could you give the variable names `wth` and `psn` slightly more descriptive names?  
* Squash the commits
* Add `[Fix #18]` as a prefix to the commit message so we have a reference to the issue in the commit log.

Thanks a lot for taking this on, I never would've solved this myself as I couldn't reproduce the problem!

spirani commented:
I changed the variable names from "wth" to "overlay-width" and "psn" to "overlay-position", and squashed the commits into https://github.com/spirani/company-quickhelp/commit/a21150c3add493e10d120a542e4a2941b570a62b

expez commented:
Thanks @spirani! :+1:
